input/input_driver.c

```diff
@@ -7289,3 +7289,61 @@ void input_keyboard_event(bool down, unsigned code,
       }
    }
 }
+
+static int input_device_get_reserved_port(int vendor_id,
+      int product_id, const char *device_name)
+{
+   settings_t *settings = config_get_ptr();
+
+   int  port;
+   bool is_reserved;
+   char settings_key[256];
+   char settings_value[256];
+   int  settings_value_vendor_id;
+   int  settings_value_product_id;
+   char settings_value_device_name[256];
+   char *settings_values[MAX_USERS];
+   settings_values[0] = settings->arrays.input_player1_reserved_device;
```

Comment:
I think we should honestly dehardcode this so that we can iterate over this in the for loop below.